dags/common/utils.py

```diff
@@ -203,28 +202,36 @@ def check_dagrun_state(dagrun: DagRun, not_allowed_states=[], allowed_states=[])
     return all(states_values)
 
 
-def process_zip_file(file_bytes, file_name):
+def process_zip_file(file_bytes, file_name, only_specific_file, form_s3_key):
     file_bytes.seek(0)
     with zipfile.ZipFile(file_bytes) as zip:
         for filename in zip.namelist():
-            file_prefix = ".".join(file_name.split(".")[:-1])
+            if only_specific_file and only_specific_file not in filename:
+                continue
             zip_file_content = zip.read(filename)
+            if not form_s3_key:
+                yield (zip_file_content, filename)
+            file_prefix = ".".join(file_name.split(".")[:-1])
             s3_filename = os.path.join(file_prefix, filename)
             yield (zip_file_content, s3_filename)
 
 
-def process_tar_file(file_bytes, file_name):
+def process_tar_file(file_bytes, file_name, only_specific_file, form_s3_key):
     file_bytes.seek(0)
     with tarfile.open(fileobj=file_bytes, mode="r") as tar:
         for filename in tar.getnames():
-            file_prefix = ".".join(file_name.split(".")[:-1])
+            if only_specific_file and only_specific_file not in filename:
+                continue
             tar_file_content = tar.extractfile(filename).read()
+            if not form_s3_key:
+                yield (tar_file_content, filename)
+            file_prefix = ".".join(file_name.split(".")[:-1])
```

Comment:
we can do simplify both functions `process_zip_file` and ` process_tar_file` by just passing, `parse_only_filenames=[]` and have:

```python
if parse_only_filenames and filename not in parse_only_filenames
```

This way, will help us add more such cases in the future.

And according to the comment for ` trigger_file_processing_elsevier` we can leave as it is without the `form_s3_key` (there is a typo also 😄 )

This applies to both functions.

Reply:
I did leave parse_only_filenames=[] on purpose  not list. In this case we don't need an additional loop inside the method, because there is a case when we want to process ONLY ONE file, which is dataset file.
Also passing the kist it will force to change the logic in in trigger_file_processing_elsevier:
I am iterating through names of all the files saved in s3
https://github.com/cern-sis/workflows/blob/3a654b2cbad3274c56fbdad25852925d06ffa342/dags/elsevier/trigger_file_processing.py#L20

Then parsing the mapping (datase.xml) file
https://github.com/cern-sis/workflows/blob/3a654b2cbad3274c56fbdad25852925d06ffa342/dags/elsevier/trigger_file_processing.py#L31


I cannot leave without  form_s3_key, because it is used for saving a file in s3:
https://github.com/cern-sis/workflows/blob/3a654b2cbad3274c56fbdad25852925d06ffa342/dags/common/pull_ftp.py#L38-L42

And just then unzipping the actual  record files:
https://github.com/cern-sis/workflows/blob/3a654b2cbad3274c56fbdad25852925d06ffa342/dags/elsevier/trigger_file_processing.py#L33-L43

Passing an array will make the code harder to read and understand